sinks/gcmautoscaling/driver.go

```diff
@@ -88,29 +108,90 @@ func isNode(metric *sink_api.Point) bool {
 	return metric.Labels[sink_api.LabelContainerName.Key] == "/"
 }
 
-func (self gcmAutocalingSink) updateMachineCapacity(input []sink_api.Timeseries) {
+func isPodContainer(metric *sink_api.Point) bool {
+	return len(metric.Labels[sink_api.LabelPodName.Key]) > 0
+}
+
+func (self *gcmAutocalingSink) updateMachineCapacityAndReservation(input []sink_api.Timeseries) {
+	self.cpuCapacity = make(map[hostTime]int64)
+	self.cpuReservation = make(map[hostTime]int64)
+	self.memCapacity = make(map[hostTime]int64)
+	self.memReservation = make(map[hostTime]int64)
 	for _, entry := range input {
 		metric := entry.Point
-		if !isNode(metric) {
+		if metric.Name != "cpu/limit" && metric.Name != "memory/limit" {
 			continue
 		}
-
 		host := metric.Labels[sink_api.LabelHostname.Key]
 		value, ok := metric.Value.(int64)
-		if !ok {
+		if !ok || value < 1 {
 			continue
 		}
-		if metric.Name == "cpu/limit" {
-			self.cpuCapacity[host] = value
-		} else if metric.Name == "memory/limit" {
-			self.memCapacity[host] = value
+
+		if isNode(metric) {
+			if metric.Name == "cpu/limit" {
+				self.cpuCapacity[hostTime{host, metric.End}] = value
+			} else if metric.Name == "memory/limit" {
+				self.memCapacity[hostTime{host, metric.End}] = value
+			}
+		} else if isPodContainer(metric) {
+			if metric.Name == "cpu/limit" {
+				self.cpuReservation[hostTime{host, metric.End}] += value
+			} else if metric.Name == "memory/limit" {
+				self.memReservation[hostTime{host, metric.End}] += value
+			}
+		}
+	}
+}
+
+func getLabels(metric *sink_api.Point) map[string]string {
+	return map[string]string{
+		LabelHostname.Key:        metric.Labels[sink_api.LabelHostname.Key],
+		LabelGCEResourceID.Key:   metric.Labels[sink_api.LabelHostID.Key],
+		LabelGCEResourceType.Key: "instance",
+	}
+}
+
+func (self *gcmAutocalingSink) getNewValue(metric *sink_api.Point, ts *gcm.Timeseries) *float64 {
+	host := metric.Labels[sink_api.LabelHostname.Key]
+
+	var val float64
+	switch metric.Name {
+	case "cpu/usage":
```

Comment:
Seems like we should start using defined values for these strings. We can do in another PR if we'd like.